awkward/array/table.py

```diff
@@ -760,7 +759,7 @@ def _concatenate_axis0(cls, tables):
             if content_type == cls.numpy.ndarray:
                 concatenate = cls.numpy.concatenate
             else:
-                raise ValueError("Concatenation not implemented for columns of type " + content_type.__name__)
+                concatenate = content_type.concatenate
```

Comment:
See also the change to awkward/array/base.py: instead of checking for the existence of a concatenate method (which would have to be done in all classes, not just Table), I've put a base implementation which just supplies the `NotImplementedError`. This implementation would be overridden by all `AwkwardArray` subclasses that actually implement `concatenate`. (Eventually, all of them should, so this is a temporary thing.)

The values of the `_contents` dict are guaranteed to be either Numpy arrays or `AwkwardArray` subclasses: it's in the `contents` property setter, which is used by the constructor.